staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -275,7 +299,320 @@ func TestDecode(t *testing.T) {
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with a negative float64 value",
+			in:            hex("c1fbbff0000000000000"), // 1(-1.0_3)
+			want:          "1969-12-31T23:59:59Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with a positive infinity",
+			in:            hex("c1f97c00"), // 1(Infinity)
+			want:          "0001-01-01T00:00:00Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with a negative infinity",
+			in:            hex("c1f9fc00"), // 1(-Infinity)
+			want:          "0001-01-01T00:00:00Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with NaN",
+			in:            hex("c1f9fc00"), // 1(NaN)
+			want:          "0001-01-01T00:00:00Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name: "tag 2 positive big integer",
+			// TODO: decide how to handle arbitrary precision numeric tags (integer and floating-point)
+			in:            hex("c249010000000000000000"), // 2(18446744073709551616)
+			failureReason: "decoding cbor data tagged with 2 produces big.Int instead of []byte",
+			assertOnError: assertNilError,
+		},
+		{
+			name: "tag 3 negative big integer",
+			// TODO: decide how to handle arbitrary precision numeric tags (integer and floating-point)
+			in:            hex("c349010000000000000000"), // 3(-18446744073709551617)
+			failureReason: "decoding cbor data tagged with 3 produces big.Int instead of []byte",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 4 decimal fraction",
+			in:            hex("c48221196ab3"), // 4([-2, 27315])
+			want:          []interface{}{int64(-2), int64(27315)},
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 5 Big floats",
+			in:            hex("c5822003"), // 5([-1, 3])
+			want:          []interface{}{int64(-1), int64(3)},
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "simple value 20",
+			in:            hex("f4"), // false
+			want:          false,
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "simple value 21",
+			in:            hex("f5"), // true
+			want:          true,
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "simple value 22",
+			in:            hex("f6"), // nil
+			want:          nil,
+			assertOnError: assertNilError,
+		},
+		{
+			name: "simple value 23",
+			in:   hex("f7"), // undefined
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+
+			}),
```

Comment:
Since we don't know what this error will look like yet, how about something like:

```suggestion
			assertOnError: func(t *testing.T, e error) {
				t.Error("todo: test must be updated once it becomes possible to configure this behavior")
			}),
```